src/components/MultiGestureCanvas/index.tsx

```diff
@@ -0,0 +1,269 @@
+
+    const zoomScale = useSharedValue(1);
+
+    // Adding together zoom scale and the initial scale to fit the content into the canvas
+    // Using the minimum content scale, so that the image is not bigger than the canvas
+    // and not smaller than needed to fit
+    const totalScale = useDerivedValue(() => zoomScale.value * minContentScale, [minContentScale]);
+
+    const panTranslateX = useSharedValue(0);
+    const panTranslateY = useSharedValue(0);
+    const panGestureRef = useRef(Gesture.Pan());
+
+    const pinchScale = useSharedValue(1);
+    const pinchTranslateX = useSharedValue(0);
+    const pinchTranslateY = useSharedValue(0);
+
+    // Total offset of the content including previous translations from panning and pinching gestures
+    const offsetX = useSharedValue(0);
+    const offsetY = useSharedValue(0);
+
+    /**
+     * Stops any currently running decay animation from panning
+     */
+    const stopAnimation = useWorkletCallback(() => {
+        cancelAnimation(offsetX);
+        cancelAnimation(offsetY);
+    });
+
+    /**
+     * Resets the canvas to the initial state and animates back smoothly
+     */
+    const reset = useWorkletCallback((animated: boolean, callbackProp?: () => void) => {
+        const callback = callbackProp ?? (() => {});
+
+        pinchScale.value = 1;
+
+        stopAnimation();
+
+        pinchScale.value = 1;
+
+        if (animated) {
+            offsetX.value = withSpring(0, SPRING_CONFIG);
+            offsetY.value = withSpring(0, SPRING_CONFIG);
+            panTranslateX.value = withSpring(0, SPRING_CONFIG);
+            panTranslateY.value = withSpring(0, SPRING_CONFIG);
+            pinchTranslateX.value = withSpring(0, SPRING_CONFIG);
+            pinchTranslateY.value = withSpring(0, SPRING_CONFIG);
+            zoomScale.value = withSpring(1, SPRING_CONFIG, callback);
+            return;
+        }
+
+        offsetX.value = 0;
+        offsetY.value = 0;
+        panTranslateX.value = 0;
+        panTranslateY.value = 0;
+        pinchTranslateX.value = 0;
+        pinchTranslateY.value = 0;
+        zoomScale.value = 1;
+
+        callback();
+    });
+
+    const {singleTapGesture: basicSingleTapGesture, doubleTapGesture} = useTapGestures({
+        canvasSize,
+        contentSize,
+        minContentScale,
+        maxContentScale,
+        offsetX,
+        offsetY,
+        pinchScale,
+        zoomScale,
+        reset,
+        stopAnimation,
+        onScaleChanged,
+        onTap,
+    });
+    const singleTapGesture = basicSingleTapGesture.requireExternalGestureToFail(doubleTapGesture, panGestureRef);
+
+    const panGestureSimultaneousList = useMemo(
+        () => (pagerRef === undefined ? [singleTapGesture, doubleTapGesture] : [pagerRef as unknown as Exclude<GestureRef, number>, singleTapGesture, doubleTapGesture]),
```

Comment:
Can we avoid this type conversion?

Reply:
no, because `GestureRef` type is not generic and we want to pass the specific pager ref here:

```ts
export type GestureRef = number | GestureType | React.RefObject<GestureType | undefined> | React.RefObject<React.ComponentType | undefined>;
```